src/response.rs

```diff
@@ -41,6 +41,28 @@ impl Response {
             Response::Http(_) => unreachable!(),
         }
     }
+
+    pub fn redirect<T>(in_: &crate::base::In, next: T) -> Result<crate::Response, failure::Error>
```

Comment:
Add another method, redirect_with() which will take the status code as third parameter.